_source/_posts/2023-01-23-keycloak-kubernetes-prod.md

```diff
@@ -0,0 +1,644 @@
+                   const namespace = 'https://www.jhipster.tech';                                                         
+                   if (event.authorization) {                                                                             
+                     api.idToken.setCustomClaim('preferred_username', event.user.email);                                  
+                     api.idToken.setCustomClaim(`${namespace}/roles`, event.authorization.roles);                         
+                     api.accessToken.setCustomClaim(`${namespace}/roles`, event.authorization.roles);                     
+                   }                                                                                                      
+                 };                                                                                                       
+
+
+                 /**                                                                                                      
+                  * Handler that will be invoked when this action is resuming after an external redirect. If your         
+                  * onExecutePostLogin function does not perform a redirect, this function can be safely ignored.         
+                  *                                                                                                       
+                  * @param {Event} event - Details about the user and the context in which they are logging in.           
+                  * @param {PostLoginAPI} api - Interface whose methods can be used to change the behavior of the login.  
+                  */                                                                                                      
+                 // exports.onContinuePostLogin = async (event, api) => {                                                 
+                 // };                                                                                                    
+
+```
+
+You can list the available actions with the following command:
+
+```shell
+auth0 actions list
+```
+The output will show the deployment status of each action:
+
+```text
+=== dev-avup2laz.us.auth0.com actions
+
+  ID                                    NAME      TYPE        STATUS  DEPLOYED  
+  ee0ea308-0e50-4be3-893b-74f9ccbb3703  jhipster  post-login  built   ✗         
+```
+
+Note the `DEPLOYED` status is `x`. Go ahead and deploy it using the action ID:
+
+```shell
+auth0 actions deploy ee0ea308-0e50-4be3-893b-74f9ccbb3703
+```
+
+Once the action is deployed, you must attach it to the login flow. Sign in to Auth0, and on the left menu choose **Actions**. Then in the Flows screen, choose **Login**. On the right bottom section, choose **Custom**. Drag the **jhipster** action to the flow diagram. Then click **Apply**. The flow should look like the following:
+
+{% img blog/keycloak-kubernetes-prod/login-flow.png alt:"Custom Auth0 Login Action" width:"500" %}{: .center-image }
```

Comment:
I prefer a light version of this image, but that's probably because I'm biased against dark themes. ;)

<img width="652" alt="Screenshot 2023-06-30 at 5 11 32 PM" src="https://github.com/oktadev/okta-blog/assets/17892/487b1047-d29f-42ae-ac2b-215c5529c403">

Reply:
Replaced